src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -23,17 +25,23 @@ internal sealed class SerializableSourceText
         /// The storage location for <see cref="SourceText"/>.
         /// </summary>
         /// <remarks>
-        /// Exactly one of <see cref="Storage"/> or <see cref="Text"/> will be non-<see langword="null"/>.
+        /// Exactly one of <see cref="_storage"/> or <see cref="_text"/> will be non-<see langword="null"/>.
         /// </remarks>
-        public ITemporaryTextStorageWithName? Storage { get; }
+        private readonly ITemporaryTextStorageWithName? _storage;
 
         /// <summary>
         /// The <see cref="SourceText"/> in the current process.
         /// </summary>
         /// <remarks>
         /// <inheritdoc cref="Storage"/>
         /// </remarks>
-        public SourceText? Text { get; }
+        private readonly SourceText? _text;
+
+        /// <summary>
+        /// Weak reference to a SourceText computed from <see cref="_storage"/>.  Useful so that if multiple requests
+        /// come in for the source text, the same one can be returned as long as something is holding it alive.
+        /// </summary>
+        private readonly WeakReference<SourceText> _computedText;
```

Comment:
so thre are two routes i cna go with in this type.  One is to have computation of the text *strongly* hold onto it.  THe benefit of this is that once computed we know for certain that any subsequent calls will get that SourceText.  

The downside is that once a SerializableSourceText is *not* needed, it will be 30 seconds till it gets cleaned up.  This is a lot of time for memory for SourceTexts to grow in a pretty unbounded fashion.  e.g. many large solution-wide changes to documents would lead to lots of memory usage here until the next AssetCache purge.

The alternative, which i've taken here, is to use a WeakReference to the computed value.  This ensures at least that lots of concurrent calls will see these cached values (since they'll be kept alive by another caller) and that calls within a short interval should see these cached values.  However, once a GC comes through *if* nothing is holding onto the SourceText, it can be dumped, which seems desirable to not grow memory usage by large amounts until the purge step is run.